dimagi/utils/dates.py

```diff
@@ -303,6 +308,11 @@ def get_validation_reason(self):
             return "You can't have an end date of %s after start date of %s" % (self.enddate, self.startdate)
         elif self.startdate < datetime.datetime(1900, 01, 01) or self.enddate < datetime.datetime(1900, 01, 01):
             return "You can't use dates earlier than the year 1900"
+        elif self.max_days >= 0:
```

Comment:
since we changed the default to `None` do we also have to add that check here? otherwise I think this might crash.

would be good to add a test-case for this either way (and especially if it does fail)